Updated App, added landing page and added react-router-dom

TishShaw opened the pull request:
### **Overview**
This pull request introduces the opening screen for the app and integrates React Router DOM for efficient page routing.

### **Changes Made:**

- **Integration of React Router DOM**: Added React Router DOM to enable dynamic page routing throughout the application.

- **Modification to main.jsx**: Implemented BrowserRouter as Router in main.jsx to ensure React Router DOM functionality is accessible app-wide.

- **Update to App.jsx**: Defined the various paths used in the application to facilitate navigation. Enhanced the structure to support the addition of new routes.

- **Creation of Home.jsx**: Developed the opening screen interface.
Included essential elements such as:

     - A prominent title display.
      
     - A 'Start' button, which on activation, navigates the user to the game area.
      
     - A static 'How to' button, intended to provide guidance and instructions for new users

Commit:
Updated App, added landing page and added react-router-dom

KatrinaWright reviewed (changes requested):
I tested the branch, and it seemed to have lost the styling for the gameboard - please re add this import back into the App.jsx and then it can be merged into main

import './App.css'

LMMSWDesigner commented:
Return the statement 

import './App.css'

to App.jsx

LMMSWDesigner reviewed (approved):
Fixed css issue

Commit:
Added app.css to main

AdventureBear commented on reverse-pacman/package.json:
there is no need to add react router to this project.  Doing so is missing a key feature of reacts ability to conditionally show and hide components.  We want conditional rendering for the start screen, not a new route.

AdventureBear reviewed (changes requested):
here is no need to add react router to this project. Doing so is missing a key feature of reacts ability to conditionally show and hide components. We want conditional rendering for the start screen, not a new route.

Commit:
Updated App and Home.jsx

TishShaw commented:
I've removed the dependency on react-router-dom, incorporated App.css directly into App.jsx, and used useState hook to elevate state from home.jsx to app.jsx.

AdventureBear commented on reverse-pacman/src/main.jsx:
this is the only file that still needs to be reverted by removing the Router references.  Everything else looks and works great.

AdventureBear reviewed (changes requested):
Everything looks good except main.jsx still has the browser router references. Once those are removed it will be good to merge! Godod job.

Commit:
Remove router im main.jsx

KatrinaWright reviewed (approved):
Tested - looks great! 
Thank you for resolving those changes with the app.css and the routing! It looks so cool now!
Merging into main <3

KatrinaWright reviewed (approved):
This is a test to see if I can resolve this request through the GUI as well

LMMSWDesigner reviewed (commented):
This is fine. It works.
